src/javascripts/zendesk-api/factory.js

```diff
@@ -342,6 +343,38 @@ module.exports = function(name, key, api) {
           title: entry.title || entry.name,
         };
       },
+      zdUpsertBatchTranslations: function(resourceList) {
+        this.store('batchTranslations', resourceList);
+        this.zdUpsertTranslationNext();
+      },
+      zdUpsertTranslationNext: function() {
+        /**
+         * Take the next resource of the batchArray and try to upsert it.
+         */
+        let resources = this.store('batchTranslations');
+        if (!resources.length) {
+          // When the pull is complete, we save the translations to Zendesk.
+          this.notifyReset();
+          this.notifyInfo(`Updating Zendesk translation files. ${io.syncLength()} files left.`);
+          this.checkAsyncComplete();
+          return;
+        }
+        this.notifyReset();
+        // Get the next translation to upsert
+        let entry = resources.shift();
+        this.store('batchTranslations', resources);
+        let txResourceName = entry.resource_name;
+        let resource = this.store(txResource.key + txResourceName);
+        let completedLocales = this.completedLanguages(resource);
+
+        let msg = `Getting translation in <b>${completedLocales.length} languages</b> for resource <b>${entry.title}</b>`
+        if (resources.length > 0)
+          msg = `${msg}.</br>${resources.length} resources queued.`;
+        this.notifyInfo(msg);
+        for (var i = 0; i < completedLocales.length; i++) { // iterate through list of locales
+          this.asyncGetTxResource(txResourceName, completedLocales[i], entry.id);
```

Comment:
This one will make multiple async calls as well, that may cause further problems if they are too many. Not sure if we should address it now, I guess it depends on how many calls they are made per translation.

Reply:
I made a test with resources with variable sizes and 17 translated languages each. It went well if you consider that the time each resource took was ~15 seconds at its maximum. I agree we could use a queue here as well. Let's leave it for a future improvement.

Reply:
That's fine with me